### ISSUE TITLE
Connectivity reliability issues with Ring Keypad v2

### ISSUE THREAD

**Issue by davidcoulson** [ImSorryButWho/HomeAssistantNotes#17]
Not sure if anyone else has ran into this? We've been having intermittent issues where the keypad would flash the red 'network' light on the front and would not send/receive any zwave commands. Usually this happens in the middle of a disarm process (enter code and press checkmark key), but have also seen it when just sending a 'disarm' button command to the keypad without any other interaction.

I'm using the Aeotec Z-Stick Gen5+ Zwave controller that is <10ft away from the keypad, so not sure why it is flakey. I've also replaced the keypad and it behaves the same, so assuming it's either a firmware issue or a compatibility issue with my z-wave stick?

**Comment by tverett** [comment]
Yes mine was doing this recently. I had constant red wireless light in the top left of the keypad but i would still get speaker outputs when i armed/disarmed from the HA dashboard on the phone/computer.  No button commands etc going back to HA would cause any actions though so it was just a speaker with non-functioning buttons.  I excluded it from my zwave network using the pinhole in the back and then re-added it. Seems to have resolved the issue for now.

**Comment by brandonjones24** [comment]
I’m having this same issue with both of our keypads…frustrating to say the least. I will try excluding and re-including to see if that fixes things, but I am not holding my breath…

**Comment by davidcoulson** [comment]
> I’m having this same issue with both of our keypads…frustrating to say the least. I will try excluding and re-including to see if that fixes things, but I am not holding my breath…

I've been on the hunt for a Zigbee Frient Keypad which apparently is more reliable, but i can't seem to find it anywhere for sale.

**Comment by Utopia69** [comment]
Same problem here, 2 Ring keypads gen 2 and Aeotec Z-Stick Gen5+
I get a red network icon on the keypad from time to time.
I think it is related to the motion sensor

**Comment by Smiley1244** [comment]
I found another post on this in the Home Assistant forum, can't find the post now though. It is the motion sensor timeout, floods the Z-Wave controller. Set config parameter 26 (Motion Sensor Timeout) higher. I'm using 30 seconds without issue. The keypad will still respond to motion and light up the keypad but it will only report it to the controller once every 30 seconds. Has greatly improved the situation for me but I do still get the occasional signal light.

**Comment by bencrundwell** [comment]
Same issue here! Was about to buy another keypad, so glad I found this thread! I'll try the timeout above and if not, try excluding. Fingers crossed!

**Comment by Grosscoder** [comment]
How to configure one button on the ring keypad V2 to night mode for alarmo in home assistant?

**Comment by mundschenk-at** [comment]
@Grosscoder That should have been a forum question over at https://community.home-assistant.io/. Please don't hijack issues or threads.

**Comment by Grosscoder** [comment]
Sorry. where can I open a new issue?

**Comment by 335iguy** [comment]
Just to add here, I had a similar issue with connectivity. I found that the alarm keypad was a hop away from the controller. With my controller being relatively close to the keypad, I set the preferred route directly to the controller. Getting way less connectivity issues. I know that this may not work for some of you, but it definitely helped in this case. I'm using ZWaveJS.
